modules/internal/String.chpl

```diff
@@ -1207,14 +1208,15 @@ module String {
       if i > 0 then
         while i < localThis.buffLen && !isInitialByte(localThis.buff[i]) do
           i += 1; // in case `start` is in the middle of a multibyte character
+    
       while i < localThis.buffLen {
-        var cp: int(32);
-        var nbytes: c_int;
-        var multibytes = (localThis.buff + i): c_string;
-        var maxbytes = (localThis.buffLen - i): ssize_t;
-        qio_decode_char_buf_esc(cp, nbytes, multibytes, maxbytes);
-        yield (cp:int(32), i:byteIndex, nbytes:int);
-        i += nbytes;
+        const (decodeRet, cp, nBytes) = decodeHelp(buff=localThis.buff, 
+                                                   buffLen=localThis.buffLen, 
+                                                   offset=i, 
+                                                   allowEsc=true);
+        
+        yield (cp:int(32), (i):byteIndex, nBytes:int);
```

Comment:
That error of buff + Thisidx + `1` asided here since before indexing change the code was
and similarly indexing error arised which wasn't fixed(as indexing changes happened while PR was still active)
`yield(cp: int(32), (i+1):byteIndex, nbytes:int;`
referencing https://github.com/chapel-lang/chapel/pull/14671 for the origin of error.

Reply:
let's drop parentheses around `i`